src/connection/server_support.c

```diff
@@ -2330,21 +2330,21 @@ css_change_ha_server_state (THREAD_ENTRY * thread_p, HA_SERVER_STATE state, bool
 	  break;
 	}
       /* If log appliers have changed their state to done, go directly to active mode */
-      if (css_check_ha_log_applier_done ())
+      if (!HA_DISABLED ())
```

Comment:
I was expecting that HA_DISABLED () case is handled here as an error (also in 'force' branch of code)
We may have added this handling in rush of implementing a functional replication code.

Please investigate 'develop' branch for legacy behaviour.

Reply:
css change_ha_server_state () is possible even when HA_DISABLED () == true on develop

Reply:
I know `css_check_ha_log_applier_done ()` is now a dummy, but shouldn't it be replaced similar functionality (that returns true if all replications were applied)?

Reply:
@ribram, I think should be done in http://jira.cubrid.org/browse/CBRD-23098

Reply:
@radudoros sorry, I missunderstood the change. The functionality should be done as CBRD-23098, but please put back the dummy condition to let any reader know the transition depends on apply being done until it is properly fixed.